backend/question-service/src/questions/questions.controller.spec.ts

```diff
@@ -1,20 +1,213 @@
+import { QuestionCategory, QuestionComplexity } from './types/question.types';
+import { NotFoundException, BadRequestException, InternalServerErrorException } from '@nestjs/common';
+import mongoose from 'mongoose';
 
 describe('QuestionsController', () => {
   let controller: QuestionsController;
+  let service: QuestionsService;
+
+  const mockObjectId = new mongoose.Types.ObjectId();
+
+  const mockQuestion: Question = {
+    _id: mockObjectId,
+    title: 'Test Question',
+    description: 'This is a test question',
+    categories: [QuestionCategory.ALGORITHMS],
+    complexity: QuestionComplexity.EASY,
+  };
+
+  const mockQuestion2: Question = {
+    _id: new mongoose.Types.ObjectId(),
+    title: 'Second Test Question',
+    description: 'This is another test question',
+    categories: [QuestionCategory.DATA_STRUCTURES],
+    complexity: QuestionComplexity.MEDIUM,
+  };
+  
+
+  const mockQuestionsService = {
+    create: jest.fn(),
+    findAll: jest.fn(),
+    findById: jest.fn(),
+    update: jest.fn(),
+    remove: jest.fn(),
+  };
+
 
   beforeEach(async () => {
     const module: TestingModule = await Test.createTestingModule({
       controllers: [QuestionsController],
-      providers: [QuestionsService],
+      providers: [{
+        provide: QuestionsService,
+        useValue: mockQuestionsService,
+      },],
     }).compile();
 
     controller = module.get<QuestionsController>(QuestionsController);
+    service = module.get<QuestionsService>(QuestionsService);
   });
 
   it('should be defined', () => {
     expect(controller).toBeDefined();
   });
+
+  describe('create', () => {
+    it('should create a question', async () => {
+      const createQuestionDto: CreateQuestionDto = {
+        title: 'New Question',
+        description: 'This is a new question',
+        categories: [QuestionCategory.ALGORITHMS],
+        complexity: QuestionComplexity.EASY,
+      };
+
+      mockQuestionsService.create.mockResolvedValue(mockQuestion);
+
+      const result = await controller.create(createQuestionDto);
+
+      expect(result).toEqual(mockQuestion);
+      expect(mockQuestionsService.create).toHaveBeenCalledWith(createQuestionDto);
+    });
+
+    it('should throw InternalServerErrorException on creation failure', async () => {
+      const createQuestionDto: CreateQuestionDto = {
+        title: 'New Question',
+        description: 'This is a new question',
+        categories: [QuestionCategory.ALGORITHMS],
+        complexity: QuestionComplexity.EASY,
+      };
+
+      mockQuestionsService.create.mockRejectedValue(new Error('Database error'));
+
+      await expect(controller.create(createQuestionDto)).rejects.toThrow(InternalServerErrorException);
+    });
+  });
+
+  describe('findAll', () => {
+    it('should return an array of questions', async () => {
+      const mockQuestions = [mockQuestion, mockQuestion2];
+      mockQuestionsService.findAll.mockResolvedValue(mockQuestions);
```

Comment:
Perhaps mockResolvedValue should not have the description field present in the questions, since we aren't returning it.